src/include/duckdb/execution/operator/aggregate/physical_window.hpp

```diff
@@ -38,12 +38,15 @@ class PhysicalWindow : public PhysicalOperator {
 	void GetData(ExecutionContext &context, DataChunk &chunk, GlobalSourceState &gstate,
 	             LocalSourceState &lstate) const override;
 
+	bool IsSource() const override {
+		return true;
+	}
 	bool ParallelSource() const override {
 		return true;
 	}
 
-	bool IsOrderPreserving() const override {
-		return true;
+	OrderPreservationType SourceOrder() const override {
+		return OrderPreservationType::NO_ORDER;
```

Comment:
`PhysicalWindow` does not have batch indices, but maybe it should?

Not something to fix in this PR, but it would be nice to guarantee that the result of a window is returned with the groups together, and in order, and then this can be changed to `FIXED_ORDER`.